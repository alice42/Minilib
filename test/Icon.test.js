import React from 'react'
import 'jest-styled-components'
import { renderCreateWithTheme } from './TestUtils'
import IconMock from './svgrMock'
// import { shallow, mount } from 'enzyme';

test('Basic Icon', () => {
  const component = renderCreateWithTheme(<IconMock name={'butterfly'} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  // expect(tree).toHaveStyleRule('color', '#464650')
  // expect(tree).toHaveStyleRule('width', '2rem')
  // expect(tree).toHaveStyleRule('height', '2rem')
})
test('Custom Icon', () => {
  const component = renderCreateWithTheme(
    <IconMock name={'butterfly'} iconSize={'7rem'} iconColor={'red'} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
  // expect(tree).toHaveStyleRule('color', 'red')
  // expect(tree).toHaveStyleRule('width', '7rem')
  // expect(tree).toHaveStyleRule('height', '7rem')
})
