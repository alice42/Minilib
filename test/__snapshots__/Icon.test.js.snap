// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Icon 1`] = `
<
  name="butterfly"
/>
`;

exports[`Custom Icon 1`] = `
<
  iconColor="red"
  iconSize="7rem"
  name="butterfly"
/>
`;
